### ISSUE TITLE
Update mail examples to use endpoints.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8595]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)| @Lukasa reported|
|-|-|
|Trac ID|trac#8595|
|Type|enhancement|
|Created|2016-07-13 09:00:38Z|
|Branch|https://github.com/twisted/twisted/tree/8595-lukasa-mailexamples|

The mail examples use {connect,listen}SSL, which means that they rely on ContextFactory, which we're trying to deprecate (see [#6923](https://github.com/twisted/twisted/issues/6923)). We should change all these examples to use endpoints instead, which resolves this problem.

<details><summary>Searchable metadata</summary>

```
trac-id__8595 8595
type__enhancement enhancement
reporter__Lukasa Lukasa
priority__normal normal
milestone__None None
branch__8595_lukasa_mailexamples 8595-lukasa-mailexamples
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__None None
time__1468400438456148 1468400438456148
changetime__1468411561962047 1468411561962047
version__None None
owner__Lukasa Lukasa

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_1"></a>|@Lukasa commented|
|-|-|

In changeset 5774e7ab9520f01ceb7c4c9c296d1ccade7b1ca8
```
#!CommitTicketReference repository="" revision="5774e7ab9520f01ceb7c4c9c296d1ccade7b1ca8"
Add topfile for #8595.
```

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_2"></a>|@Lukasa commented|
|-|-|

See the patch at https://github.com/twisted/twisted/pull/323.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @adiroiban|
|-|-|

I am testing that all is ok and will review.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set owner to @Lukasa|
|-|-|

All good.

Test executed with:

```
$ python docs/mail/examples/smtpclient_tls.py -u test -p pass -f io@example.com -t nowhere@... -h smtp.gmail.com -m README.rst -p 587

$ python docs/mail/examples/imap4client.py
IMAP4 Server Hostname: imap.gmail.com
IMAP4 Server Port (the default is 143, 993 uses SSL): 993
IMAP4 Username: test
IMAP4 Password: bla

```

Please merge. Thanks!

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|GitHub set status to `closed`|
|-|-|

In changeset 87fefcb7d7c91416f96552f4c372a72e0fe6ed2b
```
#!CommitTicketReference repository="" revision="87fefcb7d7c91416f96552f4c372a72e0fe6ed2b"
Merge 8595-lukasa-mailexamples: Update mail examples with endpoints and TLS.

Author: lukasa
Reviewer: adiroiban
Fixes: #8595
```
